api/src/main/java/org/openmrs/module/poc/clinicalservice/service/ClinicalServiceServiceImpl.java

```diff
@@ -152,36 +154,93 @@ private void deleteOldObsOfClinicalService(final Encounter fetchedEncounter, fin
+			final String conceptUuid = oldObsItem.getConcept().getUuid();
+			if ((oldObsItem.getObsId() != null) && clinicalServices.contains(conceptUuid)
 			        && !ClinicalServiceServiceImpl.EXCLUDING_CONCEPTS.contains(conceptUuid)) {
-				this.obsService.voidObs(oldItemObs,
-				    Context.getMessageSourceService().getMessage("poc.global.info.voided.forupdate"));
+				this.voidObs(oldObsItem);
+				
+				if ((oldObsItem.getObsGroup() != null) && !obsGroups.contains(oldObsItem.getObsGroup())) {
+					obsGroups.add(oldObsItem.getObsGroup());
+				}
 			}
 		}
+		
+		// voiding ObsGroups if found
+		for (final Obs obs : obsGroups) {
+			this.voidObs(obs);
+		}
 	}
 	
-	private void createNewObsForClinicalService(final Encounter fetchedEncounter,
+	@SuppressWarnings("unchecked")
+	private void createNewObsForClinicalService(final Encounter encounter,
 	        final List<Map<String, Object>> lstMapNewObs) {
 		
 		try {
 			for (final Map<String, Object> obsMap : lstMapNewObs) {
 				
-				final Concept concept = Context.getConceptService().getConceptByUuid((String) obsMap.get("concept"));
-				final Obs obs = new Obs();
-				obs.setConcept(concept);
-				this.setObsValue(obs, concept, obsMap.get("value"));
-				obs.setPerson(fetchedEncounter.getPatient());
-				fetchedEncounter.addObs(obs);
+				Concept conceptGroup = null;
+				Concept concept = null;
+				Object value = null;
+				
+				final List<Map<String, Object>> groupMembers = (List<Map<String, Object>>) obsMap.get("groupMembers");
+				
+				if (groupMembers != null) {
+					
+					conceptGroup = Context.getConceptService().getConceptByUuid((String) obsMap.get("concept"));
+					
+					for (final Map<String, Object> mapGroupMember : groupMembers) {
+						
+						concept = Context.getConceptService().getConceptByUuid((String) mapGroupMember.get("concept"));
+						value = mapGroupMember.get("value");
+						this.createObsWithGroupObs(encounter, concept, conceptGroup, value);
+					}
+					
+				} else {
+					concept = Context.getConceptService().getConceptByUuid((String) obsMap.get("concept"));
+					value = obsMap.get("value");
+					encounter.addObs(this.createOnlyObs(encounter, concept, value));
+				}
 			}
 		}
 		catch (final ParseException e) {
 			throw new APIException(e);
 		}
 	}
 	
+	private void createObsWithGroupObs(final Encounter encounter, final Concept concept, final Concept conceptGroup,
+	        final Object value) throws ParseException {
+		
+		Obs obsGroup = Context.getService(PocHeuristicService.class).findObsByEncounterAndConcept(encounter,
+		    conceptGroup);
+		if (obsGroup == null) {
+			obsGroup = new Obs();
+			obsGroup.setConcept(conceptGroup);
+			obsGroup.setPerson(encounter.getPatient());
+		}
+		final Obs obsMember = this.createOnlyObs(encounter, concept, value);
+		obsGroup.addGroupMember(obsMember);
+		encounter.addObs(obsGroup);
+		this.encounterService.saveEncounter(encounter);
```

Comment:
Does the `encounter` need to be saved both here and in `updateClinicalService`?

Reply:
In the flow of creating Obs without ObsGroup the encounter will be saved(persisted, flushed) in the updateClinicalService method at the end.